hollow/src/main/java/com/netflix/hollow/api/producer/AbstractHollowProducer.java

```diff
@@ -319,6 +322,7 @@ public HollowObjectMapper getObjectMapper() {
     public boolean enablePrimaryProducer(boolean doEnable) {
         if (doEnable) {
             singleProducerEnforcer.enable();
+            cycleCountWithPrimaryStatus = 0;
```

Comment:
This resets the value even if this producer instance was already primary, so I think we need this from [here](https://github.com/Netflix/hollow/commit/8e37a2b787886bbe398395968dbfc725d08b0942)-
```
if (!singleProducerEnforcer.isPrimary()) {
            cycleNumSincePrimary = 0;
        }
```

Reply:
Thanks for the catch.I missed the part that enablePrimaryProducer() is designed to be called before _every_ runCycle() invocation. Will change it.

Reply:
Yeah with the current single producer enforcer api we could see both these usages-

```
producer.enablePrimaryProducer(true)
while(true) runCycle(...);
```

and
``` 
while(true) {
    producer.enablePrimaryProducer(true);
    runCycle(...);
}
```